tftypes/attribute_path_error.go

```diff
@@ -11,6 +11,14 @@ type AttributePathError struct {
 	err  error
 }
 
+func (a AttributePathError) Equal(o AttributePathError) bool {
+	if !a.Path.Equal(o.Path) {
+		return false
+	}
+
```

Comment:
nitpick: can we add detection of an `Equal` method on the `err` here, and call that instead of comparing `err.Error()` if it exists?

Reply:
Are you referring to using `(reflect.Type).MethodName("Equal")` or is there a standard library type to assert against?

The last check here should probably be `errors.Is()` instead of `==` anyways.

Reply:
In the meantime have added unit testing for this new method with various scenarios including `nil` errors.

Reply:
I had in mind something like this: https://github.com/google/go-cmp/blob/395a0ac549058d781199661ca2e37cc636db43e4/cmp/compare.go#L309-L319

Forgetting that it would need to be reflection-based. Forget it, probably not worth it. :)